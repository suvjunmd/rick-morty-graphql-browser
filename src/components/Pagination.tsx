interface PaginationProps {
  page: number;
  totalPages: number;
  onNavigateToPage: (page: number) => void;
}

export default function Pagination({
  page,
  totalPages,
  onNavigateToPage,
}: PaginationProps) {
  return (
    <div className="pagination">
      <button onClick={() => onNavigateToPage(page - 1)} disabled={page === 1}>
        Previous
      </button>
      <div className="page-number">
        page {page} of {totalPages}
      </div>
      <button
        onClick={() => onNavigateToPage(page + 1)}
        disabled={page === totalPages}
      >
        Next
      </button>
    </div>
  );
}
