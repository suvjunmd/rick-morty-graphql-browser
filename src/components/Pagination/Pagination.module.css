.pagination {
    display: flex;
    margin-top: 1rem;
    justify-content: center;
}

.pageNumber {
    margin-left: 0.5rem;
    margin-right: 0.5rem;
}