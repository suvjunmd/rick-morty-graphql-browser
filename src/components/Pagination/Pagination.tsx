import styles from './Pagination.module.css';

interface PaginationProps {
  page: number;
  totalPages: number;
  onNavigateToPage: (page: number) => void;
}

export default function Pagination({
  page,
  totalPages,
  onNavigateToPage,
}: PaginationProps) {
  return (
    <div className={styles.pagination}>
      <button onClick={() => onNavigateToPage(page - 1)} disabled={page === 1}>
        Previous
      </button>
      <div className={styles.pageNumber}>
        page {page} of {totalPages}
      </div>
      <button
        onClick={() => onNavigateToPage(page + 1)}
        disabled={page === totalPages}
      >
        Next
      </button>
    </div>
  );
}
