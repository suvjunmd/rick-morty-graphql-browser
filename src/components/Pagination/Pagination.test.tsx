import { expect, it } from "vitest";
import { render, screen, fireEvent } from "@testing-library/react";
import Pagination from "./Pagination";

it("should render pagination", () => {
  const { container } = render(
    <Pagination page={1} totalPages={1} onNavigateToPage={() => {}} />
  );
  expect(screen.getByText('page 1 of 1')).toBeInTheDocument()
  expect(container).toMatchSnapshot();
});

it("should render disabled Previous button on first page", () => {
  const { container } = render(
    <Pagination page={1} totalPages={10} onNavigateToPage={() => {}} />
  );
  expect(screen.getByText('Previous')).toBeDisabled()
  expect(screen.getByText('Next')).toBeEnabled()
  expect(container).toMatchSnapshot();
});   

it("should render disabled Next button on last page", () => {
  const { container } = render(
    <Pagination page={10} totalPages={10} onNavigateToPage={() => {}} />
  );
  expect(screen.getByText('Previous')).toBeEnabled()
  expect(screen.getByText('Next')).toBeDisabled()
  expect(container).toMatchSnapshot();
});   

it("should call onNavigateToPage prop when clicked on Previous button", () => {
  const handleClick = vi.fn();

  render(
    <Pagination page={10} totalPages={10} onNavigateToPage={handleClick} />
  );
  fireEvent.click(screen.getByText('Previous'));
  expect(handleClick).toHaveBeenCalledTimes(1)
});   

it("should call onNavigateToPage prop when clicked on Next button", () => {
  const handleClick = vi.fn();

  render(
    <Pagination page={1} totalPages={10} onNavigateToPage={handleClick} />
  );
  fireEvent.click(screen.getByText('Next'));
  expect(handleClick).toHaveBeenCalledTimes(1)
});  