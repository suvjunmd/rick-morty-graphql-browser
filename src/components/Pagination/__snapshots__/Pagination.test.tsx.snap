// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render disabled Next button on last page 1`] = `
<div>
  <div
    class="_pagination_4a2aa2"
  >
    <button>
      Previous
    </button>
    <div
      class="_pageNumber_4a2aa2"
    >
      page 
      10
       of 
      10
    </div>
    <button
      disabled=""
    >
      Next
    </button>
  </div>
</div>
`;

exports[`should render disabled Previous button on first page 1`] = `
<div>
  <div
    class="_pagination_4a2aa2"
  >
    <button
      disabled=""
    >
      Previous
    </button>
    <div
      class="_pageNumber_4a2aa2"
    >
      page 
      1
       of 
      10
    </div>
    <button>
      Next
    </button>
  </div>
</div>
`;

exports[`should render pagination 1`] = `
<div>
  <div
    class="_pagination_4a2aa2"
  >
    <button
      disabled=""
    >
      Previous
    </button>
    <div
      class="_pageNumber_4a2aa2"
    >
      page 
      1
       of 
      1
    </div>
    <button
      disabled=""
    >
      Next
    </button>
  </div>
</div>
`;
